pandas/io/tests/test_clipboard.py

```diff
@@ -113,3 +116,12 @@ def test_read_clipboard_infer_excel(self):
         exp = pd.read_clipboard()
 
         tm.assert_frame_equal(res, exp)
+
+    #test case for testing invalid encoding
+    def test_invalid_encoding(self):
+        data = self.data['string']
+        with tm.assertRaises(ValueError):
+            data.to_clipboard(encoding='ascii')
```

Comment:
also read_clipboard later on uses the encoding from kwargs**, if i pop the kwargs in read clipboard before this code, will that not cause any issues?


if compat.PY3:
        try:
            text = compat.bytes_to_str(
                text, encoding=(kwargs.get('encoding') or
                                get_option('display.encoding'))
            )
        except:
            pass